peekingduck/pipeline/nodes/model/yolact_edgev1/yolact_edge_files/model.py

```diff
@@ -0,0 +1,653 @@
+            )
+            self.mask_layer = nn.Conv2d(
+                out_channels, self.num_priors * self.mask_dim, **head_layer_params
+            )
+            self.bbox_extra, self.conf_extra, self.mask_extra = [
+                make_extra(x, out_channels) for x in (0, 0, 0)
+            ]
+
+        self.aspect_ratios = aspect_ratios
+        self.scales = scales
+        self.priors: Union[None, torch.Tensor] = None
+        self.last_conv_size = (0, 0)
+
+    def forward(self, inputs: torch.Tensor) -> Dict[str, torch.Tensor]:
+        """
+        Args:
+            - inputs: The convOut from a layer in the backbone network
+                 Size: [batch_size, in_channels, conv_h, conv_w])
+        Returns a tuple (bbox_coords, class_confs, mask_output, prior_boxes) with sizes
+            - bbox_coords: [batch_size, conv_h*conv_w*num_priors, 4]
+            - class_confs: [batch_size, conv_h*conv_w*num_priors, num_classes]
+            - mask_output: [batch_size, conv_h*conv_w*num_priors, mask_dim]
+            - prior_boxes: [conv_h*conv_w*num_priors, 4]
+        """
+        src = self if self.parent[0] is None else self.parent[0]
+        conv_h = inputs.size(2)
+        conv_w = inputs.size(3)
+
+        inputs = src.upfeature(inputs)
+
+        bbox_x = src.bbox_extra(inputs)
+        conf_x = src.conf_extra(inputs)
+        mask_x = src.mask_extra(inputs)
+
+        bbox = (
+            src.bbox_layer(bbox_x)
+            .permute(0, 2, 3, 1)
+            .contiguous()
+            .view(inputs.size(0), -1, 4)
+        )
+        conf = (
+            src.conf_layer(conf_x)
+            .permute(0, 2, 3, 1)
+            .contiguous()
+            .view(inputs.size(0), -1, self.num_classes)
+        )
+        mask = (
+            src.mask_layer(mask_x)
+            .permute(0, 2, 3, 1)
+            .contiguous()
+            .view(inputs.size(0), -1, self.mask_dim)
+        )
+
+        mask = torch.tanh(mask)
+
+        priors = self.make_priors(conv_h, conv_w)
+        preds = {"loc": bbox, "conf": conf, "mask": mask, "priors": priors}
+        return preds
+
+    def make_priors(self, conv_h: int, conv_w: int) -> Optional[torch.Tensor]:
+        """Priors are [center-x, center-y, width, height] where center-x and
+        center-y are the center coordinates of the box."""
+        if self.last_conv_size != (conv_w, conv_h):
+            prior_data = []
+            for j, i in product(range(conv_h), range(conv_w)):
+                # +0.5 because priors are in center-size notation
+                center_x = (i + 0.5) / conv_w
+                center_y = (j + 0.5) / conv_h
+                for scale, aspect_ratios in zip(self.scales, self.aspect_ratios):  # type: ignore
+                    for aspect_ratio in aspect_ratios:
+                        aspect_ratio = sqrt(aspect_ratio)
+                        width = scale * aspect_ratio / 550
```

Comment:
What is the meaning of the magic number `550`? Better to be a named constant to convey its intention, 'coz future devs would not understand why divide by 550.